tests/core/pyspec/eth2spec/test/phase1/sanity/test_shard_blocks.py

```diff
@@ -0,0 +1,165 @@
+from eth2spec.test.context import (
+    PHASE0,
+    always_bls,
+    expect_assertion_error,
+    spec_state_test,
+    with_all_phases_except,
+)
+from eth2spec.test.helpers.shard_block import (
+    build_shard_block,
+    sign_shard_block,
+)
+from eth2spec.test.helpers.shard_transitions import is_full_crosslink
+from eth2spec.test.helpers.state import transition_to_valid_shard_slot
+
+
+def run_shard_blocks(spec, shard_state, signed_shard_block,
+                     beacon_parent_state,
+                     validate=True, valid=True):
+    pre_shard_state = shard_state.copy()
+    yield 'pre', pre_shard_state
+    yield 'signed_shard_block', signed_shard_block
+    yield 'validate', validate
```

Comment:
I'm not certain we'll have test vectors on the `validate=False`  path. This path is a helper for building blocks but not ever what you'd run when you receive external inputs. Given that the outputs of tests _are_ external inputs and the values of post-transitions, the results (unless no post state) will always be able to validated and thus should be 

In phase 0, there are no test vectors for `validate_results=False`